shared/lib/multichain/addresses/tron.ts

```diff
@@ -0,0 +1,281 @@
+/**
+ * Standalone Tron Address Validator (Synchronous)
+ *
+ * This utility validates Tron addresses in both Base58 and Hex formats.
+ * It works in browser environments with js-sha256 library.
+ *
+ * Requirements:
+ *   Include js-sha256: <script src="https://cdn.jsdelivr.net/npm/js-sha256@0.9.0/src/sha256.min.js"></script>
+ *
+ * Usage:
+ *   import { isAddress } from './tron-address-validator';
+ *
+ *   const isValid = isAddress('TRXaddress...'); // returns true or false (NO await needed!)
+ */
+
+import { is, string } from '@metamask/superstruct';
+import { sha256 } from 'js-sha256';
```

Comment:
For @MetaMask/policy-reviewers this is introduced to keep the tron address validator syncronous. In the past `tronweb` was using the same library to validate addresses. But he dependency was to big and introducing many other duplicate dependencies with it.